File: solutions/kotlin/darts/1/src/main/kotlin/Darts.kt
import kotlin.math.sqrt
import kotlin.math.pow
object Darts {

    fun score(x: Number, y: Number): Int =
        when (sqrt(x.toDouble().pow(2) + y.toDouble().pow(2))) {
	    in 0.0..1.0 -> 10
	    in 0.0..5.0 -> 5
	    in 0.0..10.0 -> 1
	    else -> 0
	}
}

File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y
  | h <= 1 = 10
  | h <= 5 = 5
  | h <= 10 = 1
  | otherwise = 0
  where h = sqrt (x ^ 2 + y ^ 2)
